agent/bindir/libvirtqemuhook.in

```diff
@@ -61,21 +61,23 @@ def handleMigrateBegin():
     try:
         domain = parse(sys.stdin)
         for interface in domain.getElementsByTagName("interface"):
-            source = interface.getElementsByTagName("source")[0]
-            bridge = source.getAttribute("bridge")
-            if isOldStyleBridge(bridge):
-                vlanId = bridge.replace("cloudVirBr", "")
-                phyDev = getGuestNetworkDevice()
-            elif isNewStyleBridge(bridge):
-                vlanId = re.sub(r"br(\w+)-", "", bridge)
-                phyDev = re.sub(r"-(\d+)$", "" , re.sub(r"^br", "" ,bridge))
-                netlib = networkConfig()
-                if not netlib.isNetworkDev(phyDev):
+            sources = interface.getElementsByTagName("source")
+            if sources.length > 0:
+                source = interface.getElementsByTagName("source")[0]
+                bridge = source.getAttribute("bridge")
+                if isOldStyleBridge(bridge):
+                    vlanId = bridge.replace("cloudVirBr", "")
                     phyDev = getGuestNetworkDevice()
-            else:
-                continue
-            newBrName = "br" + phyDev + "-" + vlanId
-            source.setAttribute("bridge", newBrName)
+                elif isNewStyleBridge(bridge):
+                    vlanId = re.sub(r"br(\w+)-", "", bridge)
+                    phyDev = re.sub(r"-(\d+)$", "" , re.sub(r"^br", "" ,bridge))
+                    netlib = networkConfig()
+                    if not netlib.isNetworkDev(phyDev):
+                        phyDev = getGuestNetworkDevice()
+                else:
+                    continue
+                newBrName = "br" + phyDev + "-" + vlanId
+                source.setAttribute("bridge", newBrName)
         print(domain.toxml())
```

Comment:
nit: please remove print

Reply:
Print is what returns the domain XML back to libvirt - unless you are asking me to refactor this in some way?

The only addition I have made is to check the array length:

```
sources = interface.getElementsByTagName("source")
             if sources.length > 0:
```
 
 The rest is pre-existing code.

Reply:
+1 with @mlsorensen 

It is not a bug, actually we must do it.